test/e2e/common/node/pod_resize.go

```diff
@@ -887,6 +888,11 @@ func doPodResizeTests(f *framework.Framework) {
 			ginkgo.By("verifying initial pod resize policy is as expected")
 			e2epod.VerifyPodResizePolicy(newPod, tc.containers)
 
+			// Pod must be both running and must have ready condition of status true, below are needed to solve Flaky tests, CreateSync ensures only that pod is Running.
```

Comment:
That's true, 
https://github.com/kubernetes/kubernetes/blob/74b9204b6a6bd5cce1c8b7c68bd2f39caf1b7fcb/pkg/api/pod/util.go#L250-L255

but it is not checking PodRunningReady condition , this is why i have added it later.

https://github.com/kubernetes/kubernetes/blob/74b9204b6a6bd5cce1c8b7c68bd2f39caf1b7fcb/test/utils/conditions.go#L31-L45

Reply:
I was getting flaky tests without it, this is why i added it.

Reply:
It's the same condition that CreateSync is checking. Either there's a bug in one of these checks, or there's something else going on. I'm betting on the latter. Maybe you just got lucky and the test didn't flake after adding this?

Reply:
BTW, I also was hitting a bunch of flakes in these tests, and I'm working on improving the situation in https://github.com/kubernetes/kubernetes/pull/128377. There's a lot more going there than just waiting for the pod to be ready.

Reply:
Done, removed the extra check.

Reply:
@AnishShah and @tallclair you where correct. It was luck :-) tests passed removing the extra check. As you can see it fails in two last tests which i believe has something to do with the way i use the test framework so i guess a timeout in the last run ( tries to restart kubelet but end tests asynchronously ?) . If you have time would appreciate to take a look on the two failing last tests ( it is actually one per job since two runs are performed) it has been bothering me the last week.